### PR TITLE
feat(sdk): enable deeplink pairing

### PR DESCRIPTION
**Pull request by klassare**

### CODE REVIEW

**Comment by AndreasGassmann** [comment]
Thanks for opening this PR.

We are currently working on deeplinking support (including mobile and desktop wallets) internally. I haven't considered the web-deeplinks, so this PR is a nice starting point. I will most likely merge this and then integrate it into the other deeplinking logic.

Some questions regarding the web deeplinks:
- Are there currently any example websites deployed that I could use to test this?
- Am I correct to assume that this is only relevant on desktop? So this link shouldn't be displayed if you're on an Android or iOS phone?

File: src/events.ts [hunk 1]
```diff
@@ -150,11 +152,17 @@ const showQrCode = async (
...
       dataString,
       'svg'
-    )}<p>Don't know what to do with this QR code? <a href="https://docs.walletbeacon.io/supported-wallets.html" target="_blank">Learn more</a>.</p>`,
-    confirmCallback: () => undefined
+    )}<p>Connect wallet by scanning the QR code or clicking the link button <a href="https://docs.walletbeacon.io/supported-wallets.html" target="_blank">Learn&nbsp;more</a></p>`,
+    confirmCallback: () => undefined,
+    actionCallback: async () => {
+      const base58encoded = await serializer.serialize(data)
+      const uri = `web+tezos://?type=tzip10&data=${base58encoded}`
+      window.open(uri, '_blank')
```
Comment (m-kus) [inline src/events.ts:164]: Theoretically, any web app can register a custom handler, then it's probably better to:
```
var otherWindow = window.open();
otherWindow.opener = null;
otherWindow.location = uri;
```
https://medium.com/@jitbit/target-blank-the-most-underestimated-vulnerability-ever-96e328301f4c

**Comment by klassare** [comment]
> Thanks for opening this PR.
> 
> We are currently working on deeplinking support (including mobile and desktop wallets) internally. I haven't considered the web-deeplinks, so this PR is a nice starting point. I will most likely merge this and then integrate it into the other deeplinking logic.
> 
> Some questions regarding the web deeplinks:
> 
> * Are there currently any example websites deployed that I could use to test this?
> * Am I correct to assume that this is only relevant on desktop? So this link shouldn't be displayed if you're on an Android or iOS phone?

It should work for mobile, desktop and web (from desktop browser) wallets. In general, it will not work for web wallets in a mobile browser.

For testing:
https://preview.kukai.app
https://tezos.help/beacon-demo

File: src/events.ts (hunk 1 above)
Comment (klassare) [reply src/events.ts]: That web apps need to ask for permission to become protocol handler should reduce most of the risk. But I agree that it's best to remove the back reference as you suggested.

**Comment by pascuin** [comment]
@AndreasGassmann can you give us an update on the status?

**Comment by AndreasGassmann** [comment]
I will merge this now and include it in our deeplink branch (which supports mobile and desktop wallets)

**Pull request merged**